zyte_common_items/items/search_request_template.py

```diff
@@ -1,20 +1,33 @@
 from base64 import b64encode
-from typing import List, Optional
+from typing import Any, List, Optional
 from urllib.parse import quote_plus
+from warnings import warn
 
 import attrs
 import jinja2
+from jinja2 import StrictUndefined
+from jinja2.exceptions import UndefinedError
 
 from zyte_common_items.base import Item
 from zyte_common_items.components import DetailsMetadata, Header, Request
 from zyte_common_items.converters import to_metadata_optional
 
-_TEMPLATE_ENVIRONMENT = jinja2.Environment()
+_TEMPLATE_ENVIRONMENT = jinja2.Environment(undefined=StrictUndefined)
```

Comment:
Maybe we should mention this in the release notes as a bug fix. Before, `{{ foo }}` would have been silently removed, now it will raise an `UndefinedError` exception.